src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,128 @@
+package agh.ics.oop;
+
+
+import java.util.*;
+import java.util.ArrayList;
+import java.lang.Math;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+public class GrassField extends AbstractWorldMap {
+
+    private int numOfGrass;
+    private List<Grass> grassOnMap = new ArrayList<>();
+    public GrassField(int numOfGrass)
+    {
+        this.numOfGrass = numOfGrass;
+        addGrass();
+    }
+    public void objectCollision(Vector2d position)
+    {
+        Grass grassToRemove = grassOnMap.stream().filter(grass -> (grass.getPosition().equals(position))).findFirst().orElse(null);
+        if (grassToRemove != null) {
+            addSingleGrass();
+            grassOnMap.remove(grassToRemove);
+        }
+    }
+    public void addSingleGrass()
+    {
+        List<Vector2d>Combinations = new ArrayList<>();
+        for(int x=0;x<(int) Math.sqrt(numOfGrass * 10);x++)
+        {
+            for(int y=0;y<(int) Math.sqrt(numOfGrass * 10);y++)
+            {
+                Combinations.add(new Vector2d(x,y));
+            }
+        }
+        Collections.shuffle(Combinations);
+        List<Vector2d> oldPosition = new ArrayList<>();
+        for (Grass grass : grassOnMap) {
+            oldPosition.add(grass.getPosition());
+        }
+        for (Animal animal : animals) {
+            oldPosition.add(animal.getPosition());
+        }
+        Vector2d newPosition = Combinations.stream().filter(x ->{
+            return !oldPosition.contains(x);
+        }).findFirst().orElse(null);
```

Comment:
bardzo fajne rozwiązanie, ale w przypadku brzegowym się wyspie - to `orElse(null)` wydarzy się, gdy cała mapa będzie już zapełniona. Trzeba by to obsłużyć.